wp-permastructure.php

```diff
@@ -235,13 +235,17 @@ public function add_permastructs( $rules ) {
 	 * custom taxonomies as well as the standard %author% etc...
 	 *
 	 * @param string $post_link The post URL
-	 * @param object $post      The post object
+	 * @param WP_Post $post      The post object
 	 * @param bool $leavename Passed to pre_post_link filter
 	 * @param bool $sample    Used in admin if generating an example permalink
 	 *
 	 * @return string    The parsed permalink
 	 */
-	public function parse_permalinks( $post_link, WP_Post $post, $leavename, $sample = false ) {
+	public function parse_permalinks( $post_link, $post, $leavename, $sample = false ) {
+		// Yoast Sitemap plug-in doesn't pass a WP_Post object causing a fatal, so we'll check for it and return.

```

Comment:
@roborourke - I was including changes from the perceived original plugin author: https://github.com/interconnectit/wp-permastructure